pkg/server/server.go

```diff
@@ -266,7 +266,7 @@ func (s *ProxyServer) removeBackend(agentID string, conn agent.AgentService_Conn
 }
 
 func (s *ProxyServer) addFrontend(agentID string, connID int64, p *ProxyClientConnection) {
-	klog.V(2).InfoS("Register frontend for agent", "frontend", p, "agentID", agentID, "connectionID", connID)
```

Comment:
Why are we dropping tracking of the frontend (KAS) from these logs? 

I agree we should probably use an id and not the whole structure. However the agentID is the ID for the backend. The connection ID is the ID of the individual connection. Not sure we have such an ID today but we could add one when we get the initial DIAL REQ (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/pkg/server/server.go#L417)

Reply:
Alternately we could enhance the client library to generate a UUID and send it but thats a bit trickier to do in a backward compatible way.

Reply:
Where kube-apiserver and proxy-server are 1:1 it doesn't seem like much value, but I suppose other configurations would like a frontend identifier here.

I like your second suggestion. Let me modify this PR to move in that direction.

/hold